src/app/dataset/workflow/workflow-form-field-harvest/workflow-form-field-harvest.component.html

```diff
@@ -0,0 +1,94 @@
+<div
+  class="show-on-hover"
+  *ngIf="workflowForm.get('pluginHARVEST')!.disabled || !workflowForm.get('pluginHARVEST')!.value"
+>
+  <span class="content" [innerHTML]="getImportSummary()"></span>
+</div>
+<div
+  class="form-group"
+  *ngIf="!workflowForm.get('pluginHARVEST')!.disabled && workflowForm.get('pluginHARVEST')!.value"
+>
+  <label>{{ 'harvestprotocol' | translate }} *</label>
+  <div class="form-group-radios" [formGroup]="workflowForm">
+    <label>
+      <input
+        type="radio"
+        formControlName="pluginType"
+        value="OAIPMH_HARVEST"
+        [checked]="!isProtocolHTTP()"
+        (click)="onFieldChanged('pluginHARVEST')"
+      />
+      {{ 'oai' | translate }}
+    </label>
+    <label>
+      <input
+        type="radio"
+        formControlName="pluginType"
+        value="HTTP_HARVEST"
+        [checked]="isProtocolHTTP()"
+      />
+      {{ 'http' | translate }}
+    </label>
+  </div>
+</div>
+
+<ng-container
+  *ngIf="!workflowForm.get('pluginHARVEST')!.disabled && workflowForm.get('pluginHARVEST')!.value"
+>
+  <div class="form-group" *ngIf="!isProtocolHTTP()" [formGroup]="workflowForm">
+    <label for="harvest-url">{{ 'harvesturl' | translate }} *</label>
+    <input id="harvest-url" type="text" formControlName="harvestUrl" />
+    <div
+      class="messages"
+      *ngIf="
+        workflowForm.get('harvestUrl')!.errors &&
+        workflowForm.get('harvestUrl')!.dirty &&
+        workflowForm.get('harvestUrl')!.errors!.validParameter
+      "
+    >
+      <div class="error-message">{{ 'errorharvestparameters' | translate }}</div>
+    </div>
+    <div
+      class="messages"
+      *ngIf="
+        workflowForm.get('harvestUrl')!.errors &&
+        workflowForm.get('harvestUrl')!.dirty &&
+        workflowForm.get('harvestUrl')!.errors!.validUrl
+      "
+    >
+      <div class="error-message">{{ 'errorharvesturl' | translate }}</div>
+    </div>
```

Comment:
There is a repetition of code, the only difference is the check on the validParameter or validUrl and the error message itself. Can this be written shorter? 

Something like: 
` <div
      class="messages"
      *ngIf="
        workflowForm.get('harvestUrl')!.errors &&
        workflowForm.get('harvestUrl')!.dirty
      "
    >
      <div class="error-message" *ngIf="workflowForm.get('harvestUrl')!.errors!.validParameter>{{ 'errorharvestparameters' | translate }}</div>
<div class="error-message" *ngIf="workflowForm.get('harvestUrl')!.errors!. validUrl >{{ 'errorharvesturl' | translate }}</div>
    </div>`

Reply:
Done (in both places)